VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -109,12 +116,60 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 createAssessmentTable(db);
             }
 
+            if (oldVersion < 6) {
+                Cursor cursor = db.query(false, DbStructure.Table.DOWNLOADS,
+                        new String[]{DbStructure.Column.ID, DbStructure.Column.USERNAME,
+                                DbStructure.Column.FILEPATH}, null, null, null, null, null, null);
+                if (cursor != null) {
+                    final File appExternalDir = mContext.getExternalFilesDir(null).getParentFile();
```

Comment:
This statement should either be moved to before the query, or inside the try block; we should guarantee the closing of the `Cursor` in all cases after a successful query.

Also note that the `getExternalFilesDir()` method can return `null` if the external storage is not mounted. So I suppose you should check this before the query, and only perform the migration if it returns a non-`null` value.